k9mail/src/main/java/com/fsck/k9/controller/MessagingController.java

```diff
@@ -899,8 +895,9 @@ Upload any local messages that are marked as PENDING_UPLOAD (Drafts, Sent, Trash
                     for (MessagingListener l : getListeners(listener)) {
                         l.synchronizeMailboxHeadersProgress(account, folder, headerProgress.get(), messageCount);
                     }
-                    Message localMessage = localUidMap.get(thisMess.getUid());
-                    if (localMessage == null || !localMessage.olderThan(earliestDate)) {
+                    Long localMessageDateRaw = localUidMap.get(thisMess.getUid());
+                    Date localMessageDate = localMessageDateRaw != null ? new Date(localMessageDateRaw) : null;
+                    if (localMessageDate == null || !localMessageDate.before(earliestDate)) {
```

Comment:
`earliestDate` can be `null` which leads to a `NullPointerException` in `Date.before()`.